accepted/2022/version-selection.md

```diff
@@ -0,0 +1,449 @@
+
+Considerations for the future:
+
+- We could add a `dotnet install` command to enable private acqusition.
+- We could install .NET (local) Tools into the repo-resident `.dotnet` directory instead of into the user profile location. That approach might avoid the need for a local tools manifest.
+- It would be straightforward to build a version manager (like [DNVM](https://www.codeproject.com/Articles/1005145/DNVM-DNX-and-DNU-Understanding-the-ASP-NET-Runtime)) using symlinks for the `.dotnet` directory. This would be particularly useful for preview and nightly builds.
+
+## SDK version configuration
+
+SDK version selection is (currently) exclusively controlled via `global.json`. This is a file that can be placed anywhere on your machine (although typically within a repo). The `dotnet` host will look for it in the current directory or one of its (grand-)parents, including up to the root of the drive.
+
+```json
+{
+  "sdk": {
+    "version": "6.0.201",
+    "rollForward": "latestMajor",
+    "allowPrerelease": false
+  }
+}
+```
+
+This `global.json` example requires the 6.0.201 SDK or higher, and won't accept using preview versions (like .NET 7 Preview 1). There are other `global.json` configuration options, however, the three listed are the relevant ones.
+
+### Default SDK roll-forward behavior
+
+> Customer value: This feature satisfies the "intuitive version selection" and "Convenient and consistent infra builds"customer scenarios.
+
+By default, SDK version selection does not roll-forward past a feature band. For example, the following `global.json` file requires the use of .NET 7.0.100 and will accept any version through 7.0.199. It will not allow for the use of 7.0.200 (if 7.0.1xx builds are absent).
+
+```json
+{
+  "sdk": {
+    "version": "7.0.100",
+  }
+}
+```
+
+The feature bands have a reason to exist. With each feature band (`.200`, `.300`, ...), there is typically a significant update in a tools component like Roslyn, MSBuild, or NuGet. There may be compatibility breaks. However, if there are significant breaks, that's probably something that needs to resolve. The feature bands are not intended for intentional breaking changes.
+
+This default behavior is overly strict. Starting with .NET 7, the specified version will be treated as the floor version, with no ceiling. The example above would be equivalent to the following:
+
+```json
+{
+  "sdk": {
+    "version": "7.0.100",
+    "rollForward": "latestMajor",    
+  }
+}
+```
+
+The pre-.NET 7 "pinning" behavior becomes opt-in and can be achieved with the following pattern:
+
+```json
+{
+  "sdk": {
+    "version": "7.0.100",
+    "rollForward": "feature",
+  }
+}
+```
+
+This change will only affect `global.json` files with a `version` of `7.0.0` or greater. It will not affect `global.json` files with `6.0.0` (for example) even if .NET 7 SDK is installed on the machine.
```

Comment:
Do you mean 7.0.100 or greater since this is the SDK behavior?

Reply:
Good call.